internal/code/alias.go

```diff
@@ -15,3 +28,29 @@ func Unalias(t types.Type) types.Type {
 	}
 	return types.Unalias(t)
 }
+
+// isAliasInternal checks if an alias type path is declared for a type within
+// an internal package. A best-effort attempt is made to mirror the Go internal
```

Comment:
Can you site your source for the Go internal visibility rules here?

Reply:
I'm still trying to see if there is a better source for this, but I translated the logic of the function used by the compiler: [disallowInternal](https://github.com/golang/go/blob/adf21a9b56f3498385152d701ddd0e8679fe57ed/src/cmd/go/internal/load/pkg.go#L1466) (but removing aspects that don't pertain to just the import path, since it considers more context about the packages and file system).

It seems silly but I couldn't find anything in the language spec or Go module reference, and I haven't found anything yet in the official [references docs](https://go.dev/doc/#references). I know it is deliberate behavior, because it couldn't be all us Go developers just hallucinating the existence of `internal` :joy:.

Reply:
That seems probably the best you are going to get!